helm/charts/hydra/files/automigrate.sh

```diff
@@ -0,0 +1,40 @@
+set -e
+apk add curl netcat-openbsd
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+chmod +x kubectl
+mv kubectl /usr/local/bin/kubectl
+
+[[ -f /etc/migration/migration.json ]] && kubectl delete cm "${NAME_ROOT}-migration" -n "${NAMESPACE}"
+
+# DSN is always in the form DSN=DB_TYPE://DB_USER:PASSWORD@DB_URL/DB_NAME?sslmode=disable
+# Extract DB_URL by cutting between @ and first /
+DB_URL_PORT=$(echo $DSN | cut -d '@' -f2 | cut -d '/' -f 1 )
+DB_TYPE=$(echo $DSN | cut -d ':' -f 1)
+# DB_URL is expected to be mydb.mynamespace.svc.cluster.local:1234, but the port can be optional
+# Check if it given by looking for :
+if echo "$DB_URL_PORT" | grep -q ':'; then
+  # Split URL and port by :
+  DB_URL=$(echo $DB_URL_PORT | cut -d ':' -f 1)
+  DB_PORT=$(echo $DB_URL_PORT | cut -d ':' -f 2)
+else
+  # Use the full url, since no port was given
+  DB_URL=$DB_URL_PORT
+  case $DB_TYPE in
+  	"postgres" )
+		DB_PORT="5432"
+  		;;
+  	* )
+		DB_PORT="80"
+		;;
+  esac
+fi
+
+# If port was given, we use it, if not, empty var is expanded to 0
+until nc -zv -w 5 $DB_URL $DB_PORT; do
+  echo "$DB_URL not yet ready"
+  sleep 5
+done
+
+hydra migrate sql -e --yes --config /etc/config/config.yaml
+echo {\"timestamp\": \"$(date "+%Y%m%d-%H%M%S")\",\"completed\": true} > /tmp/migration.json
+kubectl create configmap "${NAME_ROOT}-migration" -n "${NAMESPACE}" --from-file=/tmp/migration.json
```

Comment:
maybe add a retry here to do not retry everything if last CM create fails in the first try? let's assume migration did work but `kubectl` receives for example `connection refused` or `timeout`, then we will repeat the entire operation which will be unnecessary time consuming, and we may create artificial load on db unnecessary even though cli will tell us that migration was already done.